README.md

```diff
@@ -7,59 +7,411 @@
+    // of incoming transfers.
+    ctx.state.maxUnsecured = opts.maxUnsecured
+  },
+
+  // this function is called every time an incoming transfer has been prepared.
+  // throwing an error will stop the incoming transfer from being emitted as an
+  // event.
+  handleIncomingPrepare: async function (ctx, transfer) {
+    // we get the incomingFulfilledAndPrepared because it represents the most
+    // that can be owed to us, if all prepared transfers get fulfilled. The
+    // 'transfer' has already been applied to this balance.
+    const incoming = await ctx.transferLog.getIncomingFulfilledAndPrepared()
+    const bestClaim = await ctx.state.bestClaim.getMax() || { value: '0', data: null }
+
+    // make sure that if all incoming transfers are fulfilled (including the
+    // new one), it won't put us too far from the best incoming claim we've
+    // gotten.  'incoming - bestClaim.value' is the amount that our peer can
+    // default on, so it's important we limit it.
+    const exceeds = new BigNumber(incoming)
+      .subtract(bestClaim.value)
+      .greaterThan(ctx.state.maxUnsecured)
+
+    if (exceeds) {
+      throw new NotAcceptedError(transfer.id + ' exceeds max unsecured balance')
+    }
+  },
+
+  // this function is called whenever the outgoingBalance changes by a
+  // significant amount.  It may or may not be called on every transfer's
+  // fulfill, so it should not rely on being part of a payment flow.
+  createOutgoingClaim: async function (ctx, outgoingBalance) {
+    // create a claim for the total outgoing balance. This call is idempotent,
+    // because it's relating to the absolute amount owed, and doesn't modify
+    // anything.
+    const claim = Network.createClaim(outgoingBalance)
+
+    // return an object with the claim and the amount that the claim is for.
+    // this will be passed into your peer's handleIncomingClaim function.
+    return {
+      balance: outgoingBalance,
+      claim: claim
+    }
+  },
+
+  // this function is called right after the peer calls createOutgoingClaim.
+  handleIncomingClaim: async function (ctx, claim) {
+    const { balance, claim } = claim
+
+    if (Network.verify(claim, balance)) {
+      // if the incoming claim is valid and it's better than your previous best
+      // claim, set the bestClaim to the new one. If you already have a better
+      // claim this will leave it intact. It's important to use the backend's
+      // maxValueTracker here, because it will be safe across many processes.
+      await ctx.state.bestClaim.setIfMax({ value: balance, data: claim })
+    }
+  },
+
+  // called on plugin disconnect
+  disconnect: async function (ctx) {
+    const claim = ctx.state.bestClaim.getMax()
+    if (!claim) {
+      return
+    }
+
+    // submit the best claim before disconnecting. This is the only time we
+    // have to wait on the underlying ledger.
+    await Network.submitClaim(claim)
+  }
+})
```

Comment:
It's awesome that it would be so simple to write a payment channel plugin using this!